itensor/itensor_interface.ih

```diff
@@ -189,6 +189,20 @@ getVals(Iter it,
             "Last argument to .set method must be Real or Cplx scalar");
     }
 
+template<typename IntT,
+         typename Iter,
+         typename Int>
+auto
+getInts(Iter it,
+        Cplx & z,
+        Int const& iv)
+    -> stdx::enable_if_t<std::is_same<Int,int>::value,
+                         void>
+    {
+    static_assert(stdx::false_regardless_of<IntT>::value,
```

Comment:
I just ran into a bug where this line was causing .set to fail. The bug occurs when a user provides the new value but as an integer rather than an explicit real/cplx. So for a 3-index tensor, if one does T.set(3,2,1,72); to set the value to 72, it triggers this static_assert. I think if you take a look back at the original version of this function for the IndexVal case it's not straightforwardly generalizable by just replacing IndexVal -> int. Maybe instead the check should be that the last type in the arguments is convertible to either real or complex. The is_convertible trait could be useful here (not sure without trying it, just a suggestion): http://en.cppreference.com/w/cpp/types/is_convertible

Reply:
Yes I came across this, but wasn't sure if it was so bad to have the user have to input a real or complex and not an int as the last value. A silent error I could imagine is if T is rank 3 but the user thinks T is rank 4, but accidentally forgets the value they want to set, and types something like T.set(3,2,1,1). Maybe that case is too specific, it just looks a bit odd to me and doesn't seem unreasonable to expect the last input to be a float of some sort.

But for generality I guess it is best to allow anything convertible to real or complex.